Source/MOL.cpp

```diff
@@ -231,9 +247,137 @@ pc_compute_hyp_mol_flux_eb(
       }
 
       amrex::Real flux_tmp[NVAR] = {0.0};
-      AMREX_D_TERM(flux_tmp[UMX] = -q(iv, QPRES) * ebnorm[0];
-                   , flux_tmp[UMY] = -q(iv, QPRES) * ebnorm[1];
-                   , flux_tmp[UMZ] = -q(iv, QPRES) * ebnorm[2];)
+
+      auto eos = pele::physics::PhysicsType::eos();
+      amrex::Real qtempl[5 + NUM_SPECIES] = {0.0};
+      amrex::Real spl[NUM_SPECIES] = {0.0};
+      qtempl[R_UN] = -(AMREX_D_TERM(
+        q(iv, QU) * ebnorm[0], +q(iv, QV) * ebnorm[1], +q(iv, QW) * ebnorm[2]));
+      qtempl[R_UT1] = 0.0;
+      qtempl[R_UT2] = 0.0;
+      qtempl[R_P] = q(iv, QPRES);
+      qtempl[R_RHO] = q(iv, QRHO);
+      for (int n = 0; n < NUM_SPECIES; n++) {
+        qtempl[R_Y + n] = q(iv, QFS + n);
+      }
+      amrex::Real cavg = qaux(iv, QC);
+
+      // Flip the velocity about the normal for the right state - will use
+      // left  state for remainder of right state
+      amrex::Real qtempr[5 + NUM_SPECIES] = {0.0};
+      qtempr[R_UN] = -1.0 * qtempl[R_UN];
+
+      amrex::Real eos_state_rho = qtempl[R_RHO];
+      amrex::Real eos_state_p = qtempl[R_P];
+      for (int n = 0; n < NUM_SPECIES; n++) {
+        spl[n] = qtempl[R_Y + n];
+      }
+      amrex::Real eos_state_T;
+      eos.RYP2T(eos_state_rho, spl, eos_state_p, eos_state_T);
+      amrex::Real eos_state_e;
+      eos.RTY2E(eos_state_rho, eos_state_T, spl, eos_state_e);
+      amrex::Real rhoe_l = eos_state_rho * eos_state_e;
+      amrex::Real gamc_l;
+      eos.RTY2G(eos_state_rho, eos_state_T, spl, gamc_l);
+
+      // Copy left state to right (default), except normal velocity which has
+      // already been flipped
+      qtempr[R_RHO] = qtempl[R_RHO];
+      qtempr[R_UT1] = qtempl[R_UT1];
+      qtempr[R_UT2] = qtempl[R_UT2];
+      qtempr[R_P] = qtempl[R_P];
+      amrex::Real spr[NUM_SPECIES] = {0.0};
+      for (int n = 0; n < NUM_SPECIES; n++) {
+        spr[n] = spl[n];
+      }
+      amrex::Real rhoe_r = rhoe_l;
+      amrex::Real gamc_r = gamc_l;
+
+      const bool do_ebfill = ProblemSpecificFunctions::ebfill(
```

Comment:
I don't like that we unnecessarily initialize all the arrays above even when do_ebfill ends up being false. How about instead of having do_ebfill be the return of the ebfill function, have it be a member of the PeleC class that gets parmparsed, so we can skip all the stuff above when false?

This code ends up being annoyingly similar to the non-EB function, but maybe just different enough that it's not possible to consolidate and reduce copy/pasted code.

Reply:
yeah... I can't remember the reason why I did some of these things. One thing I remember is that I wanted to know if a given region of the EB would be custom-filled or not. So instead of doing the test, then filling the arrays, I did all at once. There's probably a better way, let's chat about it next week.